bin/dfx-snapshot-mk-stock

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+
+print_help() {
+  cat <<-EOF
+
+	Creates a snapshot of a stock environment, for use in CI tests.
+	EOF
+}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=s long=snapshot desc="The file to save to" variable=DFX_SNAPSHOT_FILE default="stock-snsdemo-snapshot.tar.xz"
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+: Set up SNS state and create one finalized SNS
+bin/dfx-sns-demo
+
+: Stop the replica to let it persist all its state before we save the state.
+dfx stop
```

Comment:
FYI, we'll probably want to sleep about 30 seconds before this line unless we have some way to be sure that all the requests have been handled. Because if you look in the [logs](https://github.com/dfinity/nns-dapp/actions/runs/4700088373/jobs/8334327519) of nns-dapp CI it has stuff like:
```
[Canister rrkah-fqaaa-aaaaa-aaaaq-cai] Execution of proposal: 18 succeeded.
[Canister u7xn3-ciaaa-aaaaa-aaa4a-cai] Refresh_buyer_tokens for buyer 7rleq-s52u4-wganc-do3c5-6kbms-3k5wp-gao6x-kk5oy-jkvaq-7vjt5-aqe; old e8s 0; new e8s 314100000000
[Canister u7xn3-ciaaa-aaaaa-aaa4a-cai] Swap has reached ICP target of 314100000000
[Canister u7xn3-ciaaa-aaaaa-aaa4a-cai] Token swap committed; 1 direct investors and 0 community fund investors receive a total of 314100000000 out of 314100000000 (change 0);
```
which doesn't really make sense to have when using a stock snapshot.

Reply:
Ok, I have made a script that waits for the next checkpoint and stops the replica only then.  Let's see what that does to start times.

Reply:
It _looks_ promising, with an 11s start time instead of the 40s start time I have become accustomed to.